code/modules/mod/mod.dm

```diff
@@ -0,0 +1,555 @@
+			"<span class='hear'>You hear metal noises.</span>")
+		open = !open
+	return TRUE
+
+/obj/item/mod/control/crowbar_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(!open)
+		to_chat(user, "<span class='warning'>ERROR: Suit panel not open.</span>")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+		return FALSE
+	if(modules.len)
+		for(var/obj/item/mod/module/module as anything in modules)
+			if(module.removable)
+				uninstall(module)
+				module.forceMove(drop_location())
+			else
+				audible_message("<span class='warning'>[src] indicates that [module] cannot be removed.</span>", "<span class='warning'>[src] flashes an error message.</span>")
+				playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+				return
+		I.play_tool_sound(src, 100)
+		return TRUE
+	to_chat(user, "<span class='warning'>ERROR: There's no modules on [src]!</span>")
+	playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+	return FALSE
+
+/obj/item/mod/control/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(istype(attacking_item, /obj/item/mod/module))
+		if(!open)
+			audible_message("<span class='warning'>[src] indicates that it needs to be open before installing [attacking_item].</span>", "<span class='warning'>[src] flashes an error message.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		install(attacking_item, FALSE)
+		return TRUE
+	else if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(!open)
+			audible_message("<span class='warning'>[src] indicates that it needs to be open before installing [attacking_item].</span>", "<span class='warning'>[src] flashes an error message.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		if(cell)
+			audible_message("<span class='warning'>[src] indicates that there is a cell already installed.</span>", "<span class='warning'>[src] flashes an error message.</span>")
+			playsound(src, 'sound/machines/scanbuzz.ogg', 25, TRUE)
+			return FALSE
+		attacking_item.forceMove(src)
+		cell = attacking_item
+		audible_message("<span class='notice'>[src] indicates that [cell] has been succesfully installed.</span>", "<span class='warning'>[src] flashes a confirmation message.</span>")
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		return TRUE
+	else if(is_wire_tool(attacking_item) && open)
+		wires.interact(user)
+		return TRUE
+	else if(istype(attacking_item, /obj/item/mod/paint) && paint(user, attacking_item))
+		to_chat(user, "<span class='notice'>You paint [src] with [attacking_item].</span>")
+		qdel(attacking_item)
+		return TRUE
+	else if(open && attacking_item.GetID())
+		update_access(attacking_item)
+		return TRUE
+	return ..()
+
+/obj/item/mod/control/get_cell()
+	if(!wearer)
+		return cell
+
+/obj/item/mod/control/emag_act(mob/user)
+	locked = !locked
+	to_chat(user, "<span class='notice'>You emag [src], [locked ? "locking" : "unlocking"] it.</span>")
```

Comment:
aren't emags generally only supposed to unlock stuff, not lock it?

Reply:
like, mulebot code is an exception, not the rule, IIRC

Reply:
i just wanted some emag function it doesnt really matter that much